lib/request.js

```diff
@@ -40,6 +42,18 @@ module.exports = (options) => {
             });
         });
         request.on('error', reject);
+        request.on('timeout', () => {
+            request.abort();
+            reject(new Error('socket timeout'));
```

Comment:
Yeah, it would be helpful. And I think it's same for `request.on('error', reject)`. It would be nice to add information about request in `client.js` because it gets all errors from `request.js` anyway.